core/src/main/java/org/apache/iceberg/rest/requests/RenameTableRequest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest.requests;
+
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.rest.RESTRequest;
+
+/**
+ * A REST request to rename a table.
+ */
+public class RenameTableRequest implements RESTRequest {
+
+  private TableIdentifier source;
+  private TableIdentifier destination;
+
+  @SuppressWarnings("unused")
+  public RenameTableRequest() {
+    // Needed for Jackson Deserialization.
+  }
+
+  private RenameTableRequest(TableIdentifier source, TableIdentifier destination) {
+    this.source = source;
+    this.destination = destination;
+    validate();
+  }
+
+  public void validate() {
+    Preconditions.checkArgument(source != null, "Invalid source table: null");
+    Preconditions.checkArgument(destination != null, "Invalid destination table: null");
+    Preconditions.checkArgument(!source.equals(destination),
+        "Invalid rename table request: source must not be the same as destination");
```

Comment:
I suppose it can be a no-op. I looked at the Glue catalog, and it doesn't specifically check for this case.

In terms of making it a no-op, should we just let the client send the request and allow the server to deal with it?

Reply:
Allowing it to go through and the server can deal with it.

This is in line with some of the other catalogs.